src/test/java/org/mule/tools/apikit/MainAppScaffolderWithExistingConfigApiSyncTest.java

```diff
@@ -65,72 +76,76 @@ public void reScaffold() throws Exception {
     ScaffoldingConfiguration configuration =
         new ScaffoldingConfiguration.Builder().withApi(parseResult.get()).build();
     ScaffoldingResult result = mainAppScaffolder.run(configuration);
-    assertTrue(result.isSuccess());
-    assertEquals(1, result.getGeneratedConfigs().size());
 
-    String expected = APIKitTools.readContents(getResourceAsStream("rescaffolding-apisync-version/v1/api.xml"));
-    String generated = APIKitTools.readContents(result.getGeneratedConfigs().get(0).getContent());
-
-    XMLUnit.setIgnoreWhitespace(true);
-    Diff diff = XMLUnit.compareXML(expected, generated);
-    assertTrue(diff.identical());
+    verifySuccessfulScaffolding(result, "rescaffolding-apisync-version/v1/api.xml");
 
     List<MuleConfig> muleConfigs = new ArrayList<>(result.getGeneratedConfigs());
 
-    // In the second ScaffoldingConfiguration, we have to include the mule config generated previously
     ScaffoldingConfiguration secondScaffoldingConfiguration =
         new ScaffoldingConfiguration.Builder().withApi(parseResult.get()).withMuleConfigurations(muleConfigs).build();
     ScaffoldingResult secondScaffoldingResult = mainAppScaffolder.run(secondScaffoldingConfiguration);
 
-    assertTrue(secondScaffoldingResult.isSuccess());
-    assertEquals(1, secondScaffoldingResult.getGeneratedConfigs().size());
-    assertEquals(result.getGeneratedConfigs().get(0), secondScaffoldingResult.getGeneratedConfigs().get(0));
+    verifySuccessfulScaffolding(secondScaffoldingResult, "rescaffolding-apisync-version/v1/api.xml");
   }
 
+  /**
+   *
+   * First it scaffolds an API normally with configuration of APIKit inside the main xml file with version of raml 0.0.1.
+   * Finally it re-scaffolds with version 0.0.2.
+   * Expected behaviour is that the new scaffolded API has version 0.0.2 in the main xml.
+   *
+   */
   @Test
-  public void reScaffoldDifferntVersions() throws Exception {
+  public void reScaffoldDifferentVersions() throws Exception {
     String raml = "api.raml";
     String ramlFolderV1 = "src/test/resources/rescaffolding-apisync-version/v1";
     String ramlFolderV2 = "src/test/resources/rescaffolding-apisync-version/v2";
-    XMLUnit.setIgnoreWhitespace(true);
-
     ScaffoldingResult result = scaffoldApiSync(raml, ramlFolderV1, ROOT_RAML_RESOURCE_URL, null);
```

Comment:
If you refactor a bit the `scaffoldApi`, then you can build and pass the path from here
```ScaffoldingResult result = scaffoldApi(RuntimeEdition.CE, ROOT_RAML_RESOURCE_URL + ramlFolderV1 + raml);```

Reply:
I tried to implement this but it would require a deeper refactor, I would rather leave it like this for now but adding the comment to the method.